pyteal/ast/abi/type.py

```diff
@@ -65,6 +65,64 @@ def __str__(self) -> str:
 TypeSpec.__module__ = "pyteal.abi"
 
 
+class DataStorageSchema(ABC):
```

Comment:
If `FrameVar`'s only end up working in the context of a subroutine, maybe they should be called `SubroutineVar`'s or `FrameParam` or `SubroutineParam`...

Reply:
The thinking makes sense to me, I would propose `AbstractVar` and `FrameVar` for now.

Reply:
actually this is a good thinking, for we should move the declaration of such storage model out of ABI, for it serves a wider range or generality.

The thinking of renaming it to `FrameVar` is that, it can help in router rewrite work, though we haven't reach yet. Although the work still need the help of subroutine, we use `FrameVar` for it suggests the data source is coming from frame pointer, which is pertinent to me.